components/annotation-services/ncr/src/main/java/org/phenotips/services/annotations/ncr/internal/NCRServlet.java

```diff
@@ -0,0 +1,184 @@
+
+    private static final String TEXT_LABEL = "text";
+
+    private static final String CONTENT_LABEL = "content";
+
+    private static final String APPLICATION_JSON = "application/json";
+
+    private static final long serialVersionUID = -8185410465881809448L;
+
+    /**
+     * Gets the service URL for the Neural Concept Recognizer.
+     *
+     * @return the service URL as string
+     */
+    public String getServiceUrl()
+    {
+        return SERVICE_URL;
+    }
+
+    /**
+     * Translates and forwards the post request from the clinical-text-analysis-extension. Writes the response expected
+     * by the clinical-text-analysis-extension.
+     *
+     * @param request the {@link HttpServletRequest request} object
+     * @param response the {@link HttpServletResponse response} object
+     */
+    @Override
+    protected void doPost(final HttpServletRequest request, final HttpServletResponse response)
+    {
+        try {
+            // Translate and forward the request to the NCR service.
+            final String text = forwardPost(request);
+            // Set content type for response.
+            response.setContentType(APPLICATION_JSON);
+            // Get response writer.
+            final PrintWriter writer = response.getWriter();
+            // Forward the retrieved and translated response.
+            writer.append(text);
+        } catch (final URISyntaxException e) {
+            LOGGER.error("Invalid URI syntax: {}", e.getMessage());
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        } catch (final MalformedURLException e) {
+            LOGGER.error("The NCR service url [{}] is invalid: {}", SERVICE_URL, e.getMessage());
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        } catch (final IOException e) {
+            LOGGER.error("Error with the request for or response from [{}]: {}", SERVICE_URL, e.getMessage());
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        } catch (final Exception e) {
+            LOGGER.error("An unexpected error occurred: {}", e.getMessage());
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    /**
+     * Translates the request parameters to the format expected by the NCR service located at {@link #SERVICE_URL}.
+     *
+     * @param request the incoming request
+     * @return a list of name-value pairs, in format expected by the NCR service
+     * @throws JsonProcessingException if there was an error converting request data to json string
+     */
+    @Nonnull
+    private String translateRequest(@Nonnull final HttpServletRequest request) throws JsonProcessingException
+    {
+        final String text = request.getParameter(CONTENT_LABEL);
+        final Map<String, String> params = new HashMap<>();
+        params.put(TEXT_LABEL, StringUtils.isNotBlank(text) ? text : StringUtils.EMPTY);
```

Comment:
you may use `StringUtils.defaultIfBlank(text, StringUtils.EMPTY)`

Reply:
Fixed